.github/workflows/cherry-pick-next-to-master.yml

```
@@ -0,0 +1,29 @@
+on:
+  pull_request:
+    branches:
+      - next
+    types: ['closed']
+
+jobs:
+  cherry_pick_to_master:
+    runs-on: ubuntu-latest
+    name: Cherry pick into master
+    permissions:
+      pull-requests: write
+      contents: write
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'needs cherry-pick') && github.event.pull_request.merged == true }}
```

```suggestion
    if: ${{ contains(github.event.pull_request.labels.*.name, 'needs cherry-pick') && !contains(github.event.pull_request.labels.*.name, 'cherry-pick') && github.event.pull_request.merged == true }}
```

Then filter out cherry-pick PRs and set `inherit_labels` to true?
The only downside is that cherry-picked PRs will have both `needs cherry-pick` and `cherry-pick` labels, but the upside is that all other labels will be applied 🙂

---

Technically this should not be a problem even without this condition, as the workflow will open PRs with the `master` as the target branch but it listens for `needs cherry-pick` label on PRs merged on `next` branch. I just wanted to avoid the confusion, anyhow, I think inheriting the labels has its value too.
We can manually remove the `needs cherry-pick` label from such PRs for now.